File: java/src/main/java/com/workshop/architecture/fitness/hexagon/inside/port/outbound/Membership.java
package com.workshop.architecture.fitness.hexagon.inside.port.outbound;

import java.time.LocalDate;
import java.util.UUID;

public record Membership(
        UUID id,
        UUID customerId,
        UUID planId,
        int planPrice,
        int planDurationInMonths,
        String status,
        String statusReason,
        LocalDate startDate,
        LocalDate endDate
) {
}

File: python/src/workshop_api/fitness/hexagon/inside/port/outbound/membership.py
from dataclasses import dataclass
from datetime import date
from uuid import UUID


@dataclass(frozen=True)
class Membership:
    id: UUID
    customer_id: UUID
    plan_id: UUID
    plan_price: int
    plan_duration_in_months: int
    status: str
    status_reason: str | None
    start_date: date
    end_date: date
